standard/attributes.md

```diff
@@ -858,6 +860,277 @@ For invocations that occur within field or event initializers, the member name u
+> ```
+>
+> If `T` is replaced by `string`, `T?` becomes a nullable annotation. If `T` is replaced by `int`, `T?` becomes an `int?`. When `Find` searches an `IEnumerable<string>`, the default return is `null`, but when `Find` searches an `IEnumerable<int>`, the default return is 0. As such, specifying the return type as `T?` isn’t appropriate. However, adding this attribute solves the problem:
+>
+> <!-- Example: {template:"code-in-class-lib", name:"MaybeNull2Attribute", replaceEllipsis:true, customEllipsisReplacements: ["return default;"]} -->
+> ```csharp
+> #nullable enable
+> [return: MaybeNull]
+> public T Find<T>(IEnumerable<T> sequence, Func<T, bool> predicate) { ... }
+> ```
+>
+> The attribute informs callers that the contract implies a non-nullable type, but the return value may actually be `null`. *end example*
+
+#### §The-MaybeNullWhen-Attribute The MaybeNullWhen attribute
+
+Specifies that a non-nullable argument may be `null` when the method returns the specified `bool` value. This is similar to the `MaybeNull` attribute (§The-MaybeNull-Attribute), but includes a parameter for the specified return value.
+
+#### §The-MemberNotNull-Attribute The MemberNotNull attribute
+
+Specifies that the given member won't be `null` when the method returns.
+
+> *Example*: A helper method may include the `MemberNotNull` attribute to list any fields that are assigned to a non-null value in that method. A compiler that analyzes constructors to determine whether all non-nullable reference fields have been initialized may then use this attribute to discover which fields have been set by those helper methods. Consider the following example:
+>
+> <!-- Example: {template:"standalone-lib", name:"MemberNotNullAttribute"} -->
+> ```csharp
+> #nullable enable
+> public class Container
+> {
+>     private string _uniqueIdentifier; // must be initialized.
+>     private string? _optionalMessage;
+>
+>     public Container()
+>     {
+>         Helper();
+>     }
+>
+>     public Container(string message)
+>     {
+>         Helper();
+>         _optionalMessage = message;
+>     }
+>
+>     [MemberNotNull(nameof(_uniqueIdentifier))]
+>     private void Helper()
+>     {
+>         _uniqueIdentifier = DateTime.Now.Ticks.ToString();
+>     }
+> }
+> ```
+>
+> Multiple field names may be given as arguments to the attribute’s constructor. *end example*
+
+#### §The-MemberNotNullWhen-Attribute The MemberNotNullWhen attribute
+
+Specifies that the listed member won't be `null` when the method returns the specified `bool` value.
+
+> *Example*: This attribute is like `MemberNotNull` (§The-MemberNotNull-Attribute) except that `MemberNotNullWhen` takes a `bool` argument. `MemberNotNullWhen` is intended for use in situations in which a helper method returns a `bool` indicating whether it initialized fields. *end example*
+
+#### §The-NotNull-Attribute The NotNull attribute
+
+Specifies that a nullable return value will never be `null`.
```

Comment:
Should the postconditional nature be emphasized?
```suggestion
Specifies that a nullable return value will never be `null` if the method returns (rather than throwing).
```